aspnetcore/performance/response-compression.md

```diff
@@ -0,0 +1,198 @@
+Use the `GzipCompressionProvider` to compress responses with GZip. This is the default compression provider if none are specified. 
+
+[!code-csharp[Main](response-compression/sample/FullSample/Startup.cs?name=snippet2&highlight=5)]
+
+You can set the compression level with the `GzipCompressionProviderOptions`. The default is for the fastest compression.
+
+Compression Level | Description
+--- | ---
+`CompressionLevel.Fastest` (default) | Compression should complete as quickly as possible, even if the resulting output is not optimally compressed.
+`CompressionLevel.NoCompression` | No compression should be performed.
+`CompressionLevel.Optimal` | Responses should be optimally compressed, even if the compression takes more time to complete.
+
+[!code-csharp[Main](response-compression/sample/FullSample/Startup.cs?name=snippet2&highlight=9-12)]
+
+#### Custom Providers
+You can create custom compression implementations with `ICompressionProvider`. The `encodingName` will reflect the `Accept-Encoding` header value that triggers `CreateStream()` method execution. Using the sample application, the client would submit a request with the `Accept-Encoding: custom` header. The middleware will use the custom compression implementation and return the response with a `Content-Encoding: custom` header. The client must be able to decompress the custom encoding in order for a custom compression implementation to work.
+
+[!code-csharp[Main](response-compression/sample/FullSample/Startup.cs?name=snippet2&highlight=6)]
+
+[!code-csharp[Main](response-compression/sample/FullSample/CustomCompressionProvider.cs?name=snippet1)]
+
+Submit a request to the sample application with the `Accept-Encoding: custom` header and observe the response headers. The `Vary` and `Content-Encoding` headers are present on the response. The response body (not shown) isn't compressed by the sample, as there is no compression implementation in the `CustomCompressionProvider` class of the sample. However, the sample shows where you would implement such a compression algorithm.
+
+![Fiddler window showing result of a request with the Accept-Encoding header and a value of custom. The Vary and Content-Encoding headers are added to the response.](response-compression/_static/request-custom-compression.png)
+
+### MIME types
+The middleware specifies a default set of MIME types for compression:
+* `text/plain`
+* `text/css`
+* `application/javascript`
+* `text/html`
+* `application/xml`
+* `text/xml`
+* `application/json`
+* `text/json`
+
+You can replace or append MIME types with the Response Compression Middleware options. Note that wildcard MIME types, such as `text/*` are not supported.
+
+The sample application adds a MIME type for `image/svg+xml` and will compress and serve the ASP.NET Core banner image (**banner.svg**). The new MIME type is added to the default MIME types with the [`.Concat()` (concatenate) method](https://msdn.microsoft.com/en-us/library/bb302894(v=vs.110).aspx).
```

Comment:
I don't think we need to call out the fact that you are using Concat, that's a detail that's clearly evident in the sample code and it's a very common API.